api/actions/views.py

```diff
@@ -82,12 +82,26 @@ def get_serializer_class(self):
 
     def get_object(self):
         action = None
-        if ReviewAction.objects.filter(_id=self.kwargs['action_id']).exists():
-            action = get_object_or_404(get_review_actions_queryset(), _id=self.kwargs['action_id'])
-        elif NodeRequestAction.objects.filter(_id=self.kwargs['action_id']).exists() or PreprintRequestAction.objects.filter(_id=self.kwargs['action_id']).exists():
+        if NodeRequestAction.objects.filter(_id=self.kwargs['action_id']).exists() or PreprintRequestAction.objects.filter(_id=self.kwargs['action_id']).exists():
             # No permissions allow for viewing RequestActions yet
             raise PermissionDenied('You do not have permission to view this Action')
-        if not action:
+
+        # Query all Action classes that aren't deleted
+        action = [
+            action.objects.filter(_id=self.kwargs['action_id']).include(
```

Comment:
Clever! However, I think there is a better way to write this that doesn't involve running the same query twice and is a bit cleaner. Also perhaps calling it action_type or action_subclass would make it more readable.